File: spring/oauth2-example-43zoo/oauth2-server/src/main/java/com/example/oauth2server/repository/UserRepository.java
package com.example.oauth2server.repository;

import java.util.Optional;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import com.example.oauth2server.entity.User;

@Repository
public interface UserRepository extends JpaRepository<User, Long> {
    Optional<User> findByName(String name);
}

File: jpa/spring-data-jpa-pagenation/src/main/kotlin/com/example/spring_data_jpa_pagenation/UserRepository.kt
package com.example.spring_data_jpa_pagenation

import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import org.springframework.data.repository.CrudRepository

interface UserRepository : CrudRepository<User, Long> {
    fun findByName(name: String): List<User>
    fun findByNameContaining(name: String): List<User>
    fun findByNameContaining(name: String, pageable: Pageable): Page<User>
}
